pg_search/src/api/operator/searchqueryinput.rs

```diff
@@ -103,12 +103,13 @@ pub fn query_input_restrict(
                 if !rte.is_null() {
                     let heaprelid = (*rte).relid;
                     let indexrel = locate_bm25_index(heaprelid)?;
+                    let relfilenode = indexrel.rd_locator.relNumber;
```

Comment:
because of the differences between pg12-15 and 16/17, probably best to have a centralized function (in `utils.rs`?) to extract the `relNumber` in the right way.